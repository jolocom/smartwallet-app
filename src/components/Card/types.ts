import { StyleProp, ViewStyle } from 'react-native'
import { ClaimEntry } from 'jolocom-lib/js/credentials/credential/types'

export interface IWithCustomStyle<T = ViewStyle> {
  customStyles?: StyleProp<T>
}

export interface IField {
  label: string
  value: ClaimEntry
}

export interface ICardProps {
  id: string
  type: string
  optionalFields: IField[]
  mandatoryFields: Array<IField | null>
  photo?: string | undefined
  highlight?: string | undefined
}

interface IOptionalFields extends IWithCustomStyle {
  lastFieldPadding?: string
}

export interface ICardComposition {
  OptionalFields: React.FC<IOptionalFields>
  DocumentHeader: React.FC
  OtherHeader: React.FC
  Highlight: React.FC
  Photo: React.FC
  Dots: React.FC<IWithCustomStyle>
}
