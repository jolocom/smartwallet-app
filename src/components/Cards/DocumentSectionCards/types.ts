import { DisplayVal } from '@jolocom/sdk/js/credentials'

interface CommonDocumentsProps {
  credentialName: string
  fields: Array<Required<DisplayVal>>
  onHandleMore: () => void
}

export interface OtherCardProps extends CommonDocumentsProps {
  credentialType: string
  logo?: string
}

export interface DocumentCardProps extends CommonDocumentsProps {
  holderName: string
  highlight?: string
  photo?: string
}
