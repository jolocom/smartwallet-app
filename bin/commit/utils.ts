import childProcess from 'child_process'

export const promisify =
  <T>(fn: (args: T) => void) =>
  (args: T): Promise<void> =>
    new Promise((res, rej) => {
      try {
        fn(args)
        res()
      } catch (e) {
        rej(e)
      }
    })

export const listStagedFiles = () =>
  childProcess.execSync(`git diff --cached --name-only --diff-filter=ACMR`)

export const stageModifiedFiles = (stagedFiles: string) =>
  childProcess.execSync(`echo "${stagedFiles}" | xargs git add`)

export const abortScript = (msg: string) => {
  console.log('')
  console.log('\x1b[0;33m%s\x1b[0m', msg)
  console.log('\x1b[0;31m%s\x1b[0m', 'Aborting')
  process.exit(1)
}

export const logStep = (msg: string) => {
  console.log('\x1b[0;34m%s\x1b[0m', `${msg}`)
}

export const spawnProcess = (
  onClose: (
    code: number | null,
    dataOutput: string,
    errorOutput: string,
  ) => void,
  cmd: string,
  args?: readonly string[],
  options?: childProcess.SpawnOptionsWithoutStdio,
) => {
  const spawnedProcess = childProcess.spawn(cmd, args, options)

  let dataOutput = ''
  let errorOutput = ''

  spawnedProcess.stdout.on('data', (data) => {
    dataOutput += data.toString() + '\n'
  })
  spawnedProcess.stderr.on('data', (error: Buffer) => {
    errorOutput += error.toString() + '\n'
  })

  spawnedProcess.stderr.pipe(process.stderr)
  spawnedProcess.stdout.pipe(process.stdout)

  spawnedProcess.on('close', (code) => onClose(code, dataOutput, errorOutput))

  return spawnedProcess
}
