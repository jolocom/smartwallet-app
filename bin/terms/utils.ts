import fetch from 'node-fetch';
import fs from 'fs';
import shell from 'shelljs'

const CURRENT_PATH = process.cwd();
const SECRETS_REPO = 'dev@hetz1.jolocom.io:poeditor-key'
const BASE_URL = 'https://api.poeditor.com/v2';
const API_KEY_LOCATION = `${CURRENT_PATH}/bin/poeditor-key/POEDITOR_TOKEN.txt`
const PROJECT_ID_LOCATION = `${CURRENT_PATH}/bin/poeditor-key/POEDITOR_SW2_PROJECT_ID.txt`

export enum Languages {
  en = 'en',
  de = 'de',
}

interface IResult  {
  [key: string]: {
    [key: string]: number
  }
}

export interface IResponse<T = IResult> {
  response: {
    status: string,
    code: string,
    message: string,
  },
  result: T
}

export const cloneSecrets = () => {
  return new Promise((res, rej) => {
    const keyExists = fs.existsSync(API_KEY_LOCATION);
    const idExists = fs.existsSync(PROJECT_ID_LOCATION);
    if(!keyExists || !idExists) {
      shell.cd(`${CURRENT_PATH}/bin`)
      shell.rm('-rf', `poeditor-key`)
      shell.exec(
        `git clone ${SECRETS_REPO} -b master`,
        {},
        (code, stdout, stderr) => {
          if (code != 0) return rej(new Error(stderr))
          return res(stdout)
        },
      )
    } else {
      res('Secrets are already present');
    }
  })
}

const readTextFile = (location: string) => {
  return new Promise<string>((res, rej) => {
    fs.readFile(location, 'utf8', (err, data) => {
      if (err) rej(err)
      else res(data.toString().trim())
    })
  })
}

const getPoeditorSecrets = async () => {
    const apiKey = await readTextFile(API_KEY_LOCATION)
    const projectId = await readTextFile(PROJECT_ID_LOCATION)
    return { apiKey, projectId }
}


export const sendPostRequest = async <T = IResponse, >(path: string, params: Record<string, string>) => {
  try {
    const { apiKey, projectId } = await getPoeditorSecrets();
    const requestParams = new URLSearchParams({
      api_token: apiKey,
      id: projectId,
      ...params
    })
    const res = await fetch(`${BASE_URL}${path}`, {
      method: 'POST',
      body: requestParams.toString(),
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded'
      }
    })
    return await res.json() as T;
  } catch(err) {
    console.error(err);
  }
}