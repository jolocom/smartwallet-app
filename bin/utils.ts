import fs from 'fs'
import { stringify as stringifyEnv, parse as parseEnv, Data } from 'envfile'
import shell from 'shelljs'

export const cloneRepo = async (remote: string, repoName: string) => {
  return new Promise<string>((res, rej) => {
    const dir = shell.tempdir()
    console.log(`${dir}/${repoName}`)
    shell.rm('-rf', `${dir}/${repoName}`)
    shell.exec(
      `git clone ${remote}:${repoName} -b master ${dir}/${repoName}`,
      {},
      (code, stdout, stderr) => {
        if (code != 0) return rej(new Error(stderr))
        return res(`${dir}/${repoName}`)
      },
    )
  })
}

export const catFile = async (path: string) => {
  return new Promise((res, rej) => {
    const dsn = shell.cat(path)
    if (dsn.stderr) rej(dsn.stderr)
    res(dsn.stdout)
  })
}

export const writeEnv = async (path: string, data: object) => {
  const env = stringifyEnv(data)

  return new Promise((res, rej) => {
    fs.writeFile(path, env, (err) => {
      if (err) rej(err)
      else res(undefined)
    })
  })
}

export const readEnv = async (path: string) => {
  return new Promise<Data>((res, rej) => {
    fs.readFile(path, 'utf8', (err, data) => {
      if (err) {
        console.log(
          '\x1b[4;31m%s\x1b[0m',
          'Error! Could not find the environmental config (.env).',
        )
      } else res(parseEnv(data))
    })
  })
}
