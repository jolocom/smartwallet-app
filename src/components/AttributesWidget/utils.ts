import { AttrsState } from '~/modules/attributes/types'
import { SelectableAttrI } from './index'

export const getMappedAttrubited = (attrs: AttrsState<string>) =>
  Object.keys(attrs).reduce((acc, val) => {
    acc[val] = attrs[val].map((attr) => ({
      val: attr,
      isSelected: false,
    }))
    return acc
  }, {} as AttrsState<SelectableAttrI>)
