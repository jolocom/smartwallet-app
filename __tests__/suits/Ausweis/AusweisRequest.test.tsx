import React from 'react'
import { act } from '@testing-library/react-hooks'
import { fireEvent, waitFor } from '@testing-library/react-native'
import nfcManager from 'react-native-nfc-manager'
import { useNavigation } from '@react-navigation/native'
import { aa2Module } from '@jolocom/react-native-ausweis'

import { AusweisFields, eIDScreens } from '~/screens/LoggedIn/eID/types'
import { mockSelectorReturn } from '../../mocks/libs/react-redux'
import { renderWithSafeArea } from '../../utils/renderWithSafeArea'
import { AusweisRequest } from '~/screens/LoggedIn/eID/components'
import { usePopStack } from '~/hooks/navigation'
import useSettings from '~/hooks/settings'
import eIDHooks from '~/screens/LoggedIn/eID/hooks'

const { SettingKeys } = jest.requireActual('../../../src/hooks/settings')
const mockedRequestData = {
  requiredFields: [
    AusweisFields.GivenNames,
    AusweisFields.FamilyName,
    AusweisFields.Address,
  ],
  optionalFields: [AusweisFields.PlaceOfBirth, AusweisFields.Nationality],
  certificateIssuerName: 'IssuerName',
  certificateIssuerUrl: 'IssuerUrl',
  providerName: 'ProviderName',
  providerUrl: 'ProviderUrl',
  providerInfo: 'ProviderInfo',
  effectiveValidityDate: '',
  expirationDate: '',
  setRequest: jest.fn(),
  resetRequest: jest.fn(),
}
const mockOpenUrl = jest.fn()

jest.mock('@react-navigation/native')
jest.mock('../../../src/hooks/navigation')
jest.mock('../../../src/hooks/settings')
jest.mock('react-native/Libraries/Linking/Linking', () => ({
  openURL: mockOpenUrl,
}))
jest.mock('../../../src/hooks/connection', () => ({
  __esModule: true,
  default: () => ({
    connected: true,
    showDisconnectedToast: jest.fn(),
    showConnectedToast: jest.fn(),
  }),
}))

describe('Ausweis review sceen', () => {
  const mockSettingsGet = jest.fn()
  const mockPopStack = jest.fn()
  const mockNavigate = jest.fn()
  beforeAll(() => {
    mockSelectorReturn({
      toasts: {
        active: null,
      },
      ausweis: {
        scannerKey: null,
      },
    })
    ;(nfcManager.isSupported as jest.Mock).mockReturnValue(true)
    ;(nfcManager.isEnabled as jest.Mock).mockReturnValue(true)
    ;(useNavigation as jest.Mock).mockReturnValue({
      navigate: mockNavigate,
    })
    ;(usePopStack as jest.Mock).mockReturnValue(mockPopStack)
    ;(useSettings as jest.Mock).mockImplementation(() => ({
      set: jest.fn(),
      get: mockSettingsGet,
    }))
    jest.spyOn(eIDHooks, 'useAusweisContext').mockReturnValue(mockedRequestData)
    jest.spyOn(eIDHooks, 'useAusweisCancelBackHandler')
  })
  afterEach(() => {
    mockNavigate.mockClear()
    mockSettingsGet.mockClear()
  })
  afterAll(() => {
    jest.resetAllMocks()
  })
  test('is displayed according to the designs', () => {
    const { toJSON } = renderWithSafeArea(<AusweisRequest />)
    expect(toJSON()).toMatchSnapshot()
  })

  test('user can proceed to the request review screen', async () => {
    mockSettingsGet.mockResolvedValue({ value: true })

    const { getByTestId } = renderWithSafeArea(<AusweisRequest />)
    /**
     * Wait for compatibility properties to be
     * fetched from the storage
     */
    await waitFor(() => {
      expect(mockSettingsGet).toBeCalledTimes(1)
      expect(mockSettingsGet).toBeCalledWith(
        SettingKeys.ausweisSkipCompatibility,
      )
    })

    const requesterLink = getByTestId('ausweis-requester-link')

    /**
     * Pressing provider link
     */
    act(() => {
      fireEvent.press(requesterLink)
    })
    expect(mockOpenUrl).toBeCalledTimes(1)
    expect(mockOpenUrl).toBeCalledWith(mockedRequestData.providerUrl)

    const reviewBtn = getByTestId('ausweis-cta-btn')
    act(() => {
      fireEvent.press(reviewBtn)
    })

    await waitFor(() => {
      /**
       * Note: eralier we mocked skipping the compatibility screen
       */
      expect(mockNavigate).toBeCalledWith(eIDScreens.RequestDetails)
    })
  })
  test('user can proceed to the compatibility check screen', async () => {
    mockSettingsGet.mockResolvedValue({ value: false })

    const { getByTestId } = renderWithSafeArea(<AusweisRequest />)
    /**
     * Wait for compatibility properties to be
     * fetched from the storage
     */
    await waitFor(() => {
      expect(mockSettingsGet).toBeCalledTimes(1)
      expect(mockSettingsGet).toBeCalledWith(
        SettingKeys.ausweisSkipCompatibility,
      )
    })

    const reviewBtn = getByTestId('ausweis-cta-btn')
    act(() => {
      fireEvent.press(reviewBtn)
    })

    await waitFor(() => {
      /**
       * Note: eralier we mocked skipping the compatibility screen
       */
      expect(mockNavigate).toBeCalledWith(eIDScreens.ReadinessCheck)
    })
  })

  test('user can cancel the flow', async () => {
    ;(aa2Module.cancelFlow as jest.Mock).mockRejectedValue('BAD_STATE')
    const { getByTestId } = renderWithSafeArea(<AusweisRequest />)

    await waitFor(() => {
      expect(mockSettingsGet).toBeCalledTimes(1)
    })
    const ignoreBtn = getByTestId('ausweis-ignore-btn')
    act(() => {
      fireEvent.press(ignoreBtn)
    })

    expect(aa2Module.cancelFlow).toHaveBeenCalled()
    expect(mockPopStack).toHaveBeenCalled()
  })
})
