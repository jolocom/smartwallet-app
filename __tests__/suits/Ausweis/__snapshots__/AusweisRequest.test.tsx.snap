// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ausweis request screen is displayed according to the designs 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.65)",
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#090909",
            "borderRadius": 20,
            "justifyContent": "center",
            "marginBottom": 10,
            "paddingBottom": 36,
            "paddingHorizontal": 20,
            "paddingTop": 48,
            "transform": Array [
              Object {
                "translateY": 1334,
              },
            ],
            "width": "96%",
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "position": "absolute",
              "top": -35,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderRadius": 35,
                  "height": 70,
                  "width": 70,
                },
                Object {
                  "backgroundColor": "#ffffff",
                },
              ]
            }
            testID="ausweis-logo"
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={70}
              bbWidth={70}
              focusable={false}
              height={70}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 70,
                    "width": 70,
                  },
                ]
              }
              vbHeight={70}
              vbWidth={70}
              width={70}
            >
              <RNSVGGroup>
                <RNSVGDefs>
                  <RNSVGCircle
                    cx={35}
                    cy={35}
                    name="prefix__a"
                    r={35}
                  />
                </RNSVGDefs>
                <RNSVGGroup
                  fill={null}
                  fillRule={0}
                  propList={
                    Array [
                      "fill",
                      "fillRule",
                    ]
                  }
                >
                  <RNSVGMask
                    fill={-1}
                    height="100%"
                    maskContentUnits={1}
                    maskTransform={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    maskUnits={0}
                    name="prefix__b"
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                    width="100%"
                    x="0%"
                    y="0%"
                  >
                    <RNSVGUse
                      height={0}
                      href="prefix__a"
                      width={0}
                      x={0}
                      y={0}
                    />
                  </RNSVGMask>
                  <RNSVGPath
                    d="M42.98 28.8c2.15 0 3.9-1.75 3.9-3.9s-1.75-3.9-3.9-3.9-3.899 1.75-3.899 3.9 1.75 3.9 3.9 3.9zm3.519 7.42a1.3 1.3 0 00-1.838 0l-4.28 4.28-9.49-11.866a1.3 1.3 0 00-1.935-.107L13 44.4v1.3c0 .728.572 1.3 1.3 1.3h41.764c.364 0 .676-.156.936-.364L46.499 36.219z"
                    fill={-13948631}
                    fillRule={1}
                    mask="prefix__b"
                    propList={
                      Array [
                        "fill",
                        "fillRule",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "paddingTop": undefined,
                "textAlign": "center",
              },
              Object {
                "color": "rgba(255,255,255,0.85)",
                "fontFamily": "TTCommons-Regular",
                "fontSize": 28,
                "letterSpacing": 0,
                "lineHeight": 32,
              },
              Object {
                "lineHeight": 28,
                "marginBottom": 8,
                "marginHorizontal": 10,
                "marginTop": 14,
              },
            ]
          }
          testID="interaction-title"
        >
          Ausweis.header
        </Text>
        <Text
          style={
            Array [
              Object {
                "paddingTop": undefined,
                "textAlign": "center",
              },
              Object {
                "color": "rgba(255, 255, 255, 0.7)",
                "fontFamily": "TTCommons-Regular",
                "fontSize": 16,
                "letterSpacing": 0,
                "lineHeight": 18,
              },
              Object {
                "paddingHorizontal": 10,
              },
            ]
          }
          testID="ausweis-interaction-description"
        >
          AusweisRequest.description{"serviceName":"ProviderName","interpolation":{"escapeValue":false}}
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="ausweis-requester-link"
        >
          <Text
            style={
              Array [
                Object {
                  "paddingTop": undefined,
                  "textAlign": "center",
                },
                Object {
                  "color": "#5251c1",
                  "fontFamily": "TTCommons-Regular",
                  "fontSize": 16,
                  "letterSpacing": 0,
                  "lineHeight": 18,
                },
                Object {
                  "marginBottom": 32,
                  "marginTop": 16,
                },
              ]
            }
          >
            ProviderUrl
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "flex": 0.7,
                  "marginRight": 12,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 8,
                  "justifyContent": "center",
                  "marginVertical": 0,
                  "opacity": 1,
                  "width": "100%",
                }
              }
              testID="ausweis-cta-btn"
            >
              <BVLinearGradient
                colors={
                  Array [
                    -7268030,
                    -3003031,
                  ]
                }
                endPoint={
                  Object {
                    "x": 1,
                    "y": 0,
                  }
                }
                locations={null}
                startPoint={
                  Object {
                    "x": 0,
                    "y": 0,
                  }
                }
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 8,
                      "justifyContent": "center",
                      "width": "100%",
                    },
                    Object {
                      "height": 45,
                    },
                  ]
                }
                testID="gradient"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#ffffff",
                        "paddingTop": undefined,
                        "textAlignVertical": "center",
                      },
                      Object {
                        "fontFamily": "TTCommons-Medium",
                      },
                      Object {
                        "fontSize": 18,
                      },
                      Object {},
                    ]
                  }
                >
                  AusweisRequest.proceedBtn
                </Text>
              </BVLinearGradient>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "flex": 0.3,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 8,
                  "justifyContent": "center",
                  "marginVertical": 0,
                  "opacity": 1,
                  "width": "100%",
                }
              }
              testID="ausweis-ignore-btn"
            >
              <View
                style={
                  Array [
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderRadius": 8,
                        "justifyContent": "center",
                        "width": "100%",
                      },
                      false,
                      Object {
                        "marginVertical": 0,
                      },
                    ],
                    Object {
                      "height": 45,
                    },
                    Object {
                      "borderColor": "rgba(143, 146, 161, 0.2)",
                      "borderRadius": 8,
                      "borderWidth": 2,
                    },
                  ]
                }
                testID="non-gradient"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#ffffff",
                        "paddingTop": undefined,
                        "textAlignVertical": "center",
                      },
                      Object {
                        "fontFamily": "TTCommons-Regular",
                        "opacity": 0.8,
                      },
                      Object {
                        "fontSize": 18,
                      },
                      Object {},
                    ]
                  }
                >
                  Interaction.cancelBtn
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`Ausweis review sceen is displayed according to the designs 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.65)",
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#090909",
            "borderRadius": 20,
            "justifyContent": "center",
            "marginBottom": 10,
            "paddingBottom": 36,
            "paddingHorizontal": 20,
            "paddingTop": 48,
            "transform": Array [
              Object {
                "translateY": 1334,
              },
            ],
            "width": "96%",
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "position": "absolute",
              "top": -35,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderRadius": 35,
                  "height": 70,
                  "width": 70,
                },
                Object {
                  "backgroundColor": "#ffffff",
                },
              ]
            }
            testID="ausweis-logo"
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={70}
              bbWidth={70}
              focusable={false}
              height={70}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 70,
                    "width": 70,
                  },
                ]
              }
              vbHeight={70}
              vbWidth={70}
              width={70}
            >
              <RNSVGGroup>
                <RNSVGDefs>
                  <RNSVGCircle
                    cx={35}
                    cy={35}
                    name="prefix__a"
                    r={35}
                  />
                </RNSVGDefs>
                <RNSVGGroup
                  fill={null}
                  fillRule={0}
                  propList={
                    Array [
                      "fill",
                      "fillRule",
                    ]
                  }
                >
                  <RNSVGMask
                    fill={-1}
                    height="100%"
                    maskContentUnits={1}
                    maskTransform={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    maskUnits={0}
                    name="prefix__b"
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                    width="100%"
                    x="0%"
                    y="0%"
                  >
                    <RNSVGUse
                      height={0}
                      href="prefix__a"
                      width={0}
                      x={0}
                      y={0}
                    />
                  </RNSVGMask>
                  <RNSVGPath
                    d="M42.98 28.8c2.15 0 3.9-1.75 3.9-3.9s-1.75-3.9-3.9-3.9-3.899 1.75-3.899 3.9 1.75 3.9 3.9 3.9zm3.519 7.42a1.3 1.3 0 00-1.838 0l-4.28 4.28-9.49-11.866a1.3 1.3 0 00-1.935-.107L13 44.4v1.3c0 .728.572 1.3 1.3 1.3h41.764c.364 0 .676-.156.936-.364L46.499 36.219z"
                    fill={-13948631}
                    fillRule={1}
                    mask="prefix__b"
                    propList={
                      Array [
                        "fill",
                        "fillRule",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "paddingTop": undefined,
                "textAlign": "center",
              },
              Object {
                "color": "rgba(255,255,255,0.85)",
                "fontFamily": "TTCommons-Regular",
                "fontSize": 28,
                "letterSpacing": 0,
                "lineHeight": 32,
              },
              Object {
                "lineHeight": 28,
                "marginBottom": 8,
                "marginHorizontal": 10,
                "marginTop": 14,
              },
            ]
          }
          testID="interaction-title"
        >
          Ausweis.header
        </Text>
        <Text
          style={
            Array [
              Object {
                "paddingTop": undefined,
                "textAlign": "center",
              },
              Object {
                "color": "rgba(255, 255, 255, 0.7)",
                "fontFamily": "TTCommons-Regular",
                "fontSize": 16,
                "letterSpacing": 0,
                "lineHeight": 18,
              },
              Object {
                "paddingHorizontal": 10,
              },
            ]
          }
          testID="ausweis-interaction-description"
        >
          AusweisRequest.description{"serviceName":"ProviderName","interpolation":{"escapeValue":false}}
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="ausweis-requester-link"
        >
          <Text
            style={
              Array [
                Object {
                  "paddingTop": undefined,
                  "textAlign": "center",
                },
                Object {
                  "color": "#5251c1",
                  "fontFamily": "TTCommons-Regular",
                  "fontSize": 16,
                  "letterSpacing": 0,
                  "lineHeight": 18,
                },
                Object {
                  "marginBottom": 32,
                  "marginTop": 16,
                },
              ]
            }
          >
            ProviderUrl
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "flex": 0.7,
                  "marginRight": 12,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 8,
                  "justifyContent": "center",
                  "marginVertical": 0,
                  "opacity": 1,
                  "width": "100%",
                }
              }
              testID="ausweis-cta-btn"
            >
              <BVLinearGradient
                colors={
                  Array [
                    -7268030,
                    -3003031,
                  ]
                }
                endPoint={
                  Object {
                    "x": 1,
                    "y": 0,
                  }
                }
                locations={null}
                startPoint={
                  Object {
                    "x": 0,
                    "y": 0,
                  }
                }
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 8,
                      "justifyContent": "center",
                      "width": "100%",
                    },
                    Object {
                      "height": 45,
                    },
                  ]
                }
                testID="gradient"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#ffffff",
                        "paddingTop": undefined,
                        "textAlignVertical": "center",
                      },
                      Object {
                        "fontFamily": "TTCommons-Medium",
                      },
                      Object {
                        "fontSize": 18,
                      },
                      Object {},
                    ]
                  }
                >
                  AusweisRequest.proceedBtn
                </Text>
              </BVLinearGradient>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "flex": 0.3,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 8,
                  "justifyContent": "center",
                  "marginVertical": 0,
                  "opacity": 1,
                  "width": "100%",
                }
              }
              testID="ausweis-ignore-btn"
            >
              <View
                style={
                  Array [
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderRadius": 8,
                        "justifyContent": "center",
                        "width": "100%",
                      },
                      false,
                      Object {
                        "marginVertical": 0,
                      },
                    ],
                    Object {
                      "height": 45,
                    },
                    Object {
                      "borderColor": "rgba(143, 146, 161, 0.2)",
                      "borderRadius": 8,
                      "borderWidth": 2,
                    },
                  ]
                }
                testID="non-gradient"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#ffffff",
                        "paddingTop": undefined,
                        "textAlignVertical": "center",
                      },
                      Object {
                        "fontFamily": "TTCommons-Regular",
                        "opacity": 0.8,
                      },
                      Object {
                        "fontSize": 18,
                      },
                      Object {},
                    ]
                  }
                >
                  Interaction.cancelBtn
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
